tests/unit/anchore_engine/services/policy_engine/engine/feeds/test_grypedb_sync.py

```diff
@@ -0,0 +1,174 @@
+import time
+from concurrent.futures import ThreadPoolExecutor
+from unittest.mock import MagicMock, Mock
+
+import pytest
+
+from anchore_engine.db import GrypeDBMetadata
+from anchore_engine.services.policy_engine.engine.feeds.grypedb_sync import (
+    GrypeDBSyncManager,
+    TooManyActiveGrypeDBs,
+)
+
+
+class TestGrypeDBSyncTask:
+    @pytest.fixture
+    def mock_query_active_dbs_with_data(self):
+        """
+        Creates factory that will mock _query_active_dbs return with params passed
+        """
+
+        def _mock_query(mocked_output):
+            mock_active_dbs = Mock(return_value=mocked_output)
+            GrypeDBSyncManager._query_active_dbs = mock_active_dbs
+
+        return _mock_query
+
+    @pytest.fixture
+    def mock_get_local_grypedb_checksum(self):
+        """
+        Creates factory that will mock _query_active_dbs return with params passed
+        """
+
+        def _mock_query(mocked_output):
+            mock_checksum = Mock(return_value=mocked_output)
+            GrypeDBSyncManager.get_local_grypedb_checksum = mock_checksum
+
+        return _mock_query
+
+    @pytest.fixture
+    def mock_calls_for_sync(
+        self, mock_query_active_dbs_with_data, mock_get_local_grypedb_checksum
+    ):
+        """
+        Provides ability to mock all class methods necessary to run a sync as a unit test
+        """
+
+        def _mock(mock_active_dbs=[], mock_local_checksum=""):
+            mock_query_active_dbs_with_data(mock_active_dbs)
+            mock_get_local_grypedb_checksum(mock_local_checksum)
+
+        return _mock
+
+    def test_no_active_grypedb(self, mock_calls_for_sync):
+        mock_calls_for_sync(
+            mock_active_dbs=[],
+            mock_local_checksum="eef3b1bcd5728346cb1b30eae09647348bacfbde3ba225d70cb0374da249277c",
+        )
+        result = GrypeDBSyncManager.run_grypedb_sync()
+
+        assert not result
```

Comment:
watch out for these types of assertions... if `result` is an empty list, dictionary, tuple, `0`, `False`, or `None` this will pass. It is better to assert what this should really be in this particular case